### PR TITLE
fix(highlight): add `Nvim{Light,Dark}Gray{1,2,3,4}` colors

### LINKED ISSUE #28096: Inconsistent spelling of grey/gray in `:h gui-colors`
### Problem

Among the GUI colour names there are various shades of grey. The spelling is inconsistent between the names defined by Vim and those defined by Neovim. Vim uses the spelling "gray" (with an "a"), Neovim uses "grey" (with an "e").

I don't have any opinion on which spelling we should use, but it should be consistent. If I had to pick my vote would go to "gray" because Vim set the precedent. Ideally both should work and produce the same result, but the manual should be consistent regardless of whether both spellings work or only one.

### Steps to reproduce

```vim
hi Herp guibg=NvimLightGrey1
hi Derp guibg=NvimLightGray1

hi Herp | hi Derp
```

The output is

```
Herp           xxx guibg=NvimLightGrey1
Derp           xxx cleared
```

So clearly the "gray" spelling does not work.

### Expected behavior

Both spelling should work. Interestingly, this seems to be the case with the Vim colour names.

```vim
hi Herp guibg=LightGrey
hi Derp guibg=LightGray

hi Herp | hi Derp
```

```
Herp           xxx guibg=LightGrey
Derp           xxx guibg=LightGray
```

They show different text, but the output looks the same on my terminal.

Screenshot_20240329_120028

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2739+ge2224a793

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Void Linux

### Terminal name/version

alacritty 0.13.2

### $TERM environment variable

alacritty

### Installation

AppImage

### PR DESCRIPTION
**Pull request by echasnovski**
Resolve #28096

@bfredl, this adds 8 elements to `color_name_table` but increasing its size by 8 elements didn't make it work, only by 9. As this is related to the recent arena PR, decided you might be interested to take a look.

### CODE REVIEW

**Review by clason — APPROVED** [review]

**Review by zeertzjq — CHANGES_REQUESTED** [review]
The table should be sorted alphabetically. `Gray` should come before `Green`,

**Comment by echasnovski** [comment]
> The table should be sorted alphabetically. `Gray` should come before `Green`,

Indeed, missed that (although tried to follow that by placing `Gray` before `Grey` :man_facepalming:).

File: src/nvim/highlight_group.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ typedef struct {
   char *name;
   RgbValue color;
 } color_name_table_T;
-extern color_name_table_T color_name_table[700];
+extern color_name_table_T color_name_table[709];
```
Comment (zeertzjq) [inline src/nvim/highlight_group.h:15]: Does 708 work now?

**Review by zeertzjq — COMMENTED** [review]
There also seems to be another mistake in the table:
```diff
diff --git a/src/nvim/highlight_group.c b/src/nvim/highlight_group.c
index ca8c2999e..0884c667f 100644
--- a/src/nvim/highlight_group.c
+++ b/src/nvim/highlight_group.c
@@ -2815,7 +2815,7 @@ color_name_table_T color_name_table[] = {
   { "LightGoldenrod2", RGB_(0xee, 0xdc, 0x82) },
   { "LightGoldenrod3", RGB_(0xcd, 0xbe, 0x70) },
   { "LightGoldenrod4", RGB_(0x8b, 0x81, 0x4c) },
-  { "LightGoldenRodYellow", RGB_(0xfa, 0xfa, 0xd2) },
+  { "LightGoldenrodYellow", RGB_(0xfa, 0xfa, 0xd2) },
   { "LightGray", RGB_(0xd3, 0xd3, 0xd3) },
   { "LightGreen", RGB_(0x90, 0xee, 0x90) },
   { "LightGrey", RGB_(0xd3, 0xd3, 0xd3) },
```

File: src/nvim/highlight_group.h (hunk 1 above)
Comment (echasnovski) [reply src/nvim/highlight_group.h:15]: That is super strange. Putting definition of new colors *before* their `***Green` counterpart in the array (to be alphabetical) is what made 708 work. At least for those new colors.

For more context:
- Have `extern color_name_table_T color_name_table[708];`.
- With the following order executing `:hi aaa guifg=NvimLightGray1 | hi aaa` shows `aaa xxx cleared`:
```
  { "NvimLightGreen", RGB_(0xb3, 0xf6, 0xc0) },
  { "NvimLightGray1", RGB_(0xee, 0xf1, 0xf8) },
  { "NvimLightGray2", RGB_(0xe0, 0xe2, 0xea) },
  { "NvimLightGray3", RGB_(0xc4, 0xc6, 0xcd) },
  { "NvimLightGray4", RGB_(0x9b, 0x9e, 0xa4) },
  { "NvimLightGrey1", RGB_(0xee, 0xf1, 0xf8) },
  { "NvimLightGrey2", RGB_(0xe0, 0xe2, 0xea) },
  { "NvimLightGrey3", RGB_(0xc4, 0xc6, 0xcd) },
  { "NvimLightGrey4", RGB_(0x9b, 0x9e, 0xa4) },
```
- But with the following ordering the previous command works:
```
  { "NvimLightGray1", RGB_(0xee, 0xf1, 0xf8) },
  { "NvimLightGray2", RGB_(0xe0, 0xe2, 0xea) },
  { "NvimLightGray3", RGB_(0xc4, 0xc6, 0xcd) },
  { "NvimLightGray4", RGB_(0x9b, 0x9e, 0xa4) },
  { "NvimLightGreen", RGB_(0xb3, 0xf6, 0xc0) },
  { "NvimLightGrey1", RGB_(0xee, 0xf1, 0xf8) },
  { "NvimLightGrey2", RGB_(0xe0, 0xe2, 0xea) },
  { "NvimLightGrey3", RGB_(0xc4, 0xc6, 0xcd) },
  { "NvimLightGrey4", RGB_(0x9b, 0x9e, 0xa4) },
```

So I guess alphabetical order is a **must** here?

Comment (zeertzjq) [reply src/nvim/highlight_group.h:15]: Yes, because of binary search.

Comment (echasnovski) [reply src/nvim/highlight_group.h:15]: Do you still prefer that "Goldenrod" is spelled consistently (with Vim) as a part of this PR?

Comment (zeertzjq) [reply src/nvim/highlight_group.h:15]: Yes. It's single-line change and there is no other PR to include it in.

Comment (echasnovski) [reply src/nvim/highlight_group.h:15]: > Yes. It's single-line change and there is no other PR to include it in.

Not quite. Vim consistently has "Goldenrod" spelled with the lower "r" (whenever "G" is an upper). But no problem.

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/highlight_group.h (hunk 1 above)
Comment (echasnovski) [reply src/nvim/highlight_group.h]: Replaced all "GoldenRod"s with "Goldenrod"s. Also checked that they can be assigned in highlight group (with 708 array size).

**Pull request merged**